File: lab2-functionalPrimer/haskell/func_as_arg.hs
--create inpFunc
inpFunc a b = [a..b] 

--Define applicatorFunc
applicatorFunc inpFunc s a b = 
    if s == 's' 
    then fromIntegral (sum (inpFunc a b))
    else fromIntegral (sum (inpFunc a b)) / fromIntegral (length (inpFunc a b))

main = do
    let result = applicatorFunc inpFunc 'a' 1 8 --Call applicatorFunc with inpFunc and 'a' as args
    putStrLn("sum = " ++ show(result))

File: lab2-functionalPrimer/python3/func_as_arg.py
def listFunc():
    return [i for i in range(a, b+1)] #Create list of ints from 1 to 5, Haskell equivalent [1..5]

def applicatorFunc(inpFunc, s, a, b):
    if s=='s':
        return sum(inpFunc(a, b))
    else:
        return sum(inpFunc(a, b))/ len(inpFunc(a, b))

print(applicatorFunc(listFunc, 's', 1, 5))
